### PR TITLE
[JIT] [Easy] Factor out non tensor peephole

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet
* #56642 Only optimize while change has been made
* #56438 [JIT] integer value refinement
* #56437 Factor out isDominatedBy
* #55978 Add Value * == Value * peephole
* **#55977 [JIT] [Easy] Factor out non tensor peephole**
* #55926 Add list len refinement
* #55925 Add handling of symbolic shapes
* #54809 Initial Symbolic Shape Analysis

Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes.

### CODE REVIEW

File: torch/csrc/jit/passes/peephole_non_tensor.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+          GRAPH_UPDATE(
+              "Removing ",
+              getHeader(node),
+              " as input type subtypes output type");
+          node->output()->replaceAllUsesWith(node->input());
+          changed = true;
+        }
+      }
+
+      // [aliasing sensitive optimizations]
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/peephole_non_tensor.cpp:120]: nitpick: is this comment still relevant?

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Review by penguinwu — APPROVED** [review]
LGTM

**Review by penguinwu — APPROVED** [review]
Did not look very closely, but LGTM

File: torch/csrc/jit/passes/peephole_non_tensor.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    return changed;
+  }
+
+ private:
+  std::shared_ptr<Graph> graph_;
+};
+
+bool PeepholeOptimizeNonTensor(const std::shared_ptr<Graph>& graph) {
+  PeepholeOptimizeNonTensorImpl peephole(graph);
+  bool changed = peephole.run();
```
Comment (desertfire) [inline torch/csrc/jit/passes/peephole_non_tensor.cpp:128]: Do we need to apply EliminateDeadCode here like in https://github.com/pytorch/pytorch/blob/72c3ee073f982c69d532435972a16bd9958413bd/torch/csrc/jit/passes/peephole.cpp#L551 ?

**Review by desertfire — COMMENTED** [review]
LGTM

**Review by desertfire — APPROVED** [review]

File: torch/csrc/jit/passes/peephole_non_tensor.cpp (hunk 2 above)
Comment (eellison) [reply torch/csrc/jit/passes/peephole_non_tensor.cpp:128]: This is being invoked as part of PeepholeOptimize which has it's own DCE call so i'm not invoking it here

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

**Pull request closed without merging**